x/evm/keeper/integration_test.go

```diff
@@ -1,246 +1,425 @@
-
-	stateBytes, err := json.MarshalIndent(genesisState, "", "  ")
-	s.Require().NoError(err)
-
-	// Initialize the chain
-	newapp.InitChain(
-		abci.RequestInitChain{
-			ChainId:         chainID,
-			Validators:      []abci.ValidatorUpdate{},
-			AppStateBytes:   stateBytes,
-			ConsensusParams: app.DefaultConsensusParams,
-		},
-	)
+		It("performs a transfer transaction", func() {
+			senderPriv := s.keyring.GetPrivKey(0)
+			receiver := s.keyring.GetKey(1)
+			txArgs := evmtypes.EvmTxArgs{
+				To:     &receiver.Addr,
+				Amount: big.NewInt(1000),
+				// Hard coded gas limit to avoid failure on gas estimation because
+				// of the param
+				GasLimit: 100000,
+			}
 
-	s.app = newapp
-	s.SetupApp(false, chainID)
-}
+			res, err := s.factory.ExecuteEthTx(senderPriv, txArgs)
+			Expect(err).To(BeNil())
+			Expect(res.IsOK()).To(Equal(true), "transaction should have succeeded", res.GetLog())
+		})
 
-func getNonce(addressBytes []byte) uint64 {
-	return s.app.EvmKeeper.GetNonce(
-		s.ctx,
-		common.BytesToAddress(addressBytes),
-	)
-}
+		It("fails when trying to perform contract deployment", func() {
+			senderPriv := s.keyring.GetPrivKey(0)
+			constructorArgs := []interface{}{"coin", "token", uint8(18)}
+			compiledContract := contracts.ERC20MinterBurnerDecimalsContract
+			contractAddr, err := s.factory.DeployContract(
+				senderPriv,
+				evmtypes.EvmTxArgs{}, // Default values
+				factory.ContractDeploymentData{
+					Contract:        compiledContract,
+					ConstructorArgs: constructorArgs,
+				},
+			)
 
-func buildEthTx(
-	priv *ethsecp256k1.PrivKey,
-	to *common.Address,
-	gasLimit uint64,
-	gasPrice *big.Int,
-	gasFeeCap *big.Int,
-	gasTipCap *big.Int,
-	accesses *ethtypes.AccessList,
-) *evmtypes.MsgEthereumTx {
-	chainID := s.app.EvmKeeper.ChainID()
-	from := common.BytesToAddress(priv.PubKey().Address().Bytes())
-	nonce := getNonce(from.Bytes())
-	data := make([]byte, 0)
-	ethTxParams := &evmtypes.EvmTxArgs{
-		ChainID:   chainID,
-		To:        to,
-		Nonce:     nonce,
-		GasLimit:  gasLimit,
-		GasPrice:  gasPrice,
-		GasTipCap: gasTipCap,
-		GasFeeCap: gasFeeCap,
-		Accesses:  accesses,
-		Input:     data,
-	}
-	msgEthereumTx := evmtypes.NewTx(ethTxParams)
-	msgEthereumTx.From = from.String()
-	return msgEthereumTx
-}
+			Expect(err).NotTo(BeNil())
+			Expect(err.Error()).To(ContainSubstring("EVM Create operation is disabled"))
+			Expect(contractAddr).To(Equal(common.Address{}))
+		})
+
+		It("performs a contract call to the staking precompile", func() {
+			senderPriv := s.keyring.GetPrivKey(0)
+			contractAddress := common.HexToAddress(staking.PrecompileAddress)
+			contractABI, err := staking.LoadPrecompileABI()
+			Expect(err).To(BeNil())
+
+			totalSupplyTxArgs := evmtypes.EvmTxArgs{
+				To: &contractAddress,
+			}
+
+			validatorAddress := s.network.GetValidators()[0].OperatorAddress
+			totalSupplyArgs := factory.CallArgs{
+				ContractABI: contractABI,
+				MethodName:  staking.ValidatorMethod,
```

Comment:
can you call a transaction (eg. delegate) instead?

Reply:
Since this is blocked by precompiles I will add this in an another PR